pkg/migrate/migrate.go

```diff
@@ -104,10 +108,63 @@ func Migrate(ctx context.Context, w *log.Logger, clientset k8sclient.Interface,
 		return fmt.Errorf("failed to scale up pods: %w", err)
 	}
 
+	if setDefaults {
+		err = swapDefaults(ctx, w, clientset, sourceSCName, destSCName)
+		if err != nil {
+			return fmt.Errorf("failed to change default StorageClass from %s to %s: %w", sourceSCName, destSCName, err)
+		}
+	}
+
 	w.Printf("\nSuccess!\n")
 	return nil
 }
 
+func swapDefaults(ctx context.Context, w *log.Logger, clientset k8sclient.Interface, oldDefaultSC string, newDefaultSC string) error {
+	// create a pod for each PVC migration, and wait for it to finish
+	w.Printf("\nChanging default StorageClass from %s to %s\n", oldDefaultSC, newDefaultSC)
+	err := mutateSC(ctx, w, clientset, oldDefaultSC, func(sc *storagev1.StorageClass) (*storagev1.StorageClass, error) {
+		// check to make sure that this is actually the default StorageClass before deleting the annotation
+		if sc.Annotations == nil {
+			return nil, fmt.Errorf("%s is not the default StorageClass", oldDefaultSC)
```

Comment:
what if this was the default but the script failed the first time before setting the new default? or is that not a case we want to handle?

Reply:
There shouldn't be a way to fail after successfully applying by my reading (anything that returns an error shouldn't update the state of the object)

Reply:
you remove the annotation from the storage class first then add to the new storage class. if you fail to add but successfully remove wont the script fail the next time with is not the default StorageClass?

its an edge case. but i thought worth noting. im not gonna block the pr for it but if you choose not to handle it now maybe add a comment?

Reply:
Added a bunch of code to handle this (no existing default StorageClass) and also the case where the desired dest StorageClass was already the default